client/app/lib/components/form/fields/TextField.jsx

```diff
@@ -6,25 +6,46 @@ import propsAreEqual from './utils/propsAreEqual';
 
 const styles = {
   textFieldStyle: { margin: '8px 10px 8px 0px' },
+  empty: { margin: '0px 0px 0px 0px' },
 };
 
 const FormTextField = (props) => {
-  const { field, fieldState, disabled, label, renderIf, ...custom } = props;
+  const {
+    field,
+    fieldState,
+    disabled,
+    label,
+    renderIf,
+    onChangeCustom,
+    margins,
+    className,
+    ...custom
+  } = props;
   if (!renderIf) {
     return null;
   }
 
   return (
     <TextField
+      className={className}
       {...field}
       disabled={disabled}
       label={label}
       error={!!fieldState.error}
+      onChange={(event) =>
+        onChangeCustom
+          ? onChangeCustom(event.target.value)
+          : field.onChange(event)
+      }
       helperText={
         fieldState.error && formatErrorMessage(fieldState.error.message)
       }
       {...custom}
-      style={styles.textFieldStyle}
+      style={margins ? styles.textFieldStyle : styles.empty}
+      // disable 'e' value typed in number field
```

Comment:
why only e though?

Reply:
In that case, we can keep the 'e' and add a comment there. Also, we shouldnt check after submitting - should be done before.